HW4_Sapozhnikov/prototool.py

```diff
@@ -0,0 +1,564 @@
+            elif score == delete_score:
+                traceback_matrix[i][j] = "delete"
+            elif score == insert_score:
+                traceback_matrix[i][j] = "insert"
+            else:
+                traceback_matrix[i][j] = "none"
+
+    # Traceback to find the aligned sequences
+    aligned_seq_on = []
+    aligned_seq2 = []
+
+    counter_identity: int = 0
+    counter_gaps: int = 0
+
+    i, j = max_i, max_j
+
+    while i > 0 and j > 0:
+        if traceback_matrix[i][j] == "match":
+            aligned_seq_on.append(seq_on[i - 1])
+            aligned_seq2.append(seq2[j - 1])
+            counter_identity += 1
+            i -= 1
+            j -= 1
+        elif traceback_matrix[i][j] == "delete":
+            aligned_seq_on.append(seq_on[i - 1])
+            aligned_seq2.append("-")
+            counter_gaps += 1
+            i -= 1
+        elif traceback_matrix[i][j] == "insert":
+            aligned_seq_on.append("-")
+            aligned_seq2.append(seq2[j - 1])
+            counter_gaps += 1
+            j -= 1
+        else:
+            break
+
+    # Reverse the aligned sequences
+    aligned_seq_on = "".join(aligned_seq_on[::-1])
+    aligned_seq2 = "".join(aligned_seq2[::-1])
+
+    alignment_length = (len(aligned_seq_on)
+                        if len(aligned_seq_on) < len(aligned_seq2)
+                        else len(aligned_seq2))
+
+    # Form an output dictionary
+    alignment_dict['aligned_seq_on'] = aligned_seq_on
+
+    identity = round(counter_identity/alignment_length, 4)
+
+    alignment_dict[f'aligned_seq{seq_id+1}'] = {'seq': aligned_seq2,
+                                                'length': alignment_length,
+                                                'score': alignment_score,
+                                                'identity': identity,
+                                                'gaps': counter_gaps}
+
+    # Prettify an alignment output
+    if prettify is True:
+        prettify_alignment(aligned_seq_on, aligned_seq2)
+    else:
+        pass
+
+    return alignment_dict
+
+
+def count_protein_molecular_weight(*seqs_list: Union[List[str], str]) -> dict:
+    """
+    :param seqs_list: seqs_list is a list of strings without whitespace
+    (e.g. 'AlaSer'). You can put as many sequences as you wish.
+    :return: This function returns molecular weight of the protein.
+    """
+    result = {}
+    for seq in seqs_list:
+        protein_weight = 0
+        aminoacids = [seq[i:i + 3] for i in range(0, len(seq), 3)]
+        for i, aminoacid in enumerate(aminoacids):
+            if aminoacid in AMINOACIDS_DICT.keys():
+                aminoacid_weight = (AMINOACIDS_DICT[aminoacid]
+                                    ['MOLECULAR_WEIGHTS'])
+                protein_weight += aminoacid_weight
+                result[seq] = protein_weight
+    return result
+
+
+def from_proteins_seqs_to_rna(*seqs_list: Union[List[str], str]) -> dict:
```

Comment:
Не глагол. Тут можно было бы что-то в духе
```suggestion
def get_protein_mrnas(*seqs_list: Union[List[str], str]) -> dict:
```